engine/src/main/java/org/camunda/bpm/engine/impl/externaltask/LockedExternalTaskImpl.java

```diff
@@ -164,6 +174,14 @@ public static LockedExternalTaskImpl fromEntity(ExternalTaskEntity externalTaskE
     result.variables = new VariableMapImpl();
     execution.collectVariables(result.variables, variablesToFetch, isLocal, deserializeVariables);
 
+    if(includeExtensionProperties) {
+      Map<String, String> properties = (Map<String, String>) execution.getActivity().getProperty(BpmnProperties.EXTENSION_PROPERTIES.getName());
+      if(properties == null) {
+        properties = Collections.EMPTY_MAP;
```

Comment:
It is suggested to use the functions rather than the constants for type-safety.

```suggestion
        properties = Collections.emptyMap();
```

Reply:
I feared you might say that. :D Although, I think it's nice to have that additional info, I agree that there seems to be rarely a use-case for it. I am fine with changing it to always return an empty map. Using the function instead of constant for that makes sense.

Reply:
I think it makes sense from a consumer's point of view. I don't really want to care if I enabled the option or not and I don't want to change my implementation with regards to that in case I disable/enable the option.